static/js/components/FavoritesButton.jsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import PropTypes from "prop-types";
+
+import IconButton from "@material-ui/core/IconButton";
+import StarIcon from "@material-ui/icons/Star";
+import StarBorderIcon from "@material-ui/icons/StarBorder";
+
+import * as Actions from "../ducks/favorites";
+
+const FavoritesButton = ({ source_id }) => {
```

Comment:
```suggestion
const FavoritesButton = ({ sourceID }) => {
```
throughout (camelCase in JS)